File: University Student Rating/Code/server/UniversityStudentRatingService/ServiceLogicLayer/Interfaces/ISearchService.cs
using DataAccessLayer.DbContext;
using ServiceLogicLayer.DTO;
using ServiceLogicLayer.Models;
using ServiceLogicLayer.Models.AccountModels;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ServiceLogicLayer.Interfaces
{
    public interface ISearchService
    {
        IEnumerable<UserProfile> SearchUsers(SearchModel model);
        IEnumerable<ConcurentResultDTO> SearchConcurents(ConcurentModel model, IStatisticService statisticService);
    }
}

File: University Student Rating/Code/androidclient/UniversityStudentRating/app/src/main/java/com/example/arturrinkis/universitystudentrating/ServerServiceAPI/Interfaces/Services/ISearchService.java
package com.example.arturrinkis.universitystudentrating.ServerServiceAPI.Interfaces.Services;

import com.example.arturrinkis.universitystudentrating.DTO.ConcurentModel;
import com.example.arturrinkis.universitystudentrating.DTO.ConcurentResult;
import com.example.arturrinkis.universitystudentrating.DTO.SearchModel;
import com.example.arturrinkis.universitystudentrating.DTO.UserProfile;

import java.util.ArrayList;

public interface ISearchService {
    ArrayList<UserProfile> searchUsers(SearchModel model);
    ArrayList<ConcurentResult> searchConcurents(ConcurentModel model);
}
